omen2/table.py

```diff
@@ -131,12 +131,27 @@ def remove(self, obj: "ObjBase" = None, **kws):
                 log.debug("not removing obj, because it is None")
                 return
             obj = self.select_one(**kws)
+
         if not obj or not obj._is_bound:
             log.debug("not removing object that isn't in the db")
             return
         assert obj._table is self
         obj._remove()
 
+    def remove_all(self, **kws):
+        """Remove all matching objects from the db."""
+        if self._in_tx():
+            for obj in self.select(**kws):
+                self._db_remove(obj)
+        else:
+            self.db.delete(self.table_name, **kws)
```

Comment:
Is thread-safety a potential issue here? E.g. thread 1 starts `remove_all()` and removes everything from the DB but not yet the cache, meanwhile thread 2 grabs an object from the cache and starts working with it.

Reply:
possibly?  i'll write a test.  probably need to engage a table lock

Reply:
wait, is that a problem?  thread 2 could have grabbed the object before the removal.  it can always work with a "deleted object".   that's not a new issue.   and commits of that object will "update nothing in the db"